crates/p2p_proto/src/common.rs

```diff
@@ -135,6 +145,30 @@ impl TryFromProtobuf<proto::common::Hash> for Hash {
     }
 }
 
+impl ToProtobuf<proto::common::Hash256> for Hash256 {
+    fn to_protobuf(self) -> proto::common::Hash256 {
+        proto::common::Hash256 {
+            elements: self.0.as_fixed_bytes().into(),
+        }
+    }
+}
+
+impl TryFromProtobuf<proto::common::Hash256> for Hash256 {
+    fn try_from_protobuf(
+        input: proto::common::Hash256,
+        field_name: &'static str,
+    ) -> Result<Self, std::io::Error> {
+        if input.elements.len() != H256::len_bytes() {
```

Comment:
Should we allow for shorter sequences too (like in Felt252)?

Reply:
Good question -- sure, we could do that, but given that this is supposed to be a hash value I think _requiring_ 32 bytes is actually reasonable. Probably should point this out in the spec though. Can we make such a requirement in the protobuf specification?

Reply:
Adding this to the spec really makes sense. I'm afraid in the protobuf spec we're only able to add a comment, there's nothing like a fixed size array in the types so we must use `bytes` internally.